include/alpaka/mem/buf/cpu/Copy.hpp

```diff
@@ -170,13 +171,13 @@ namespace alpaka
                             {
                                 meta::ndLoopIncIdx(
                                     extentWithoutInnermost,
-                                    [&](vec::Vec<DimMin1, ExtentSize> const & idx)
```

Comment:
At least fo code where I know that it will always be executed on the host only, this probably makes the code slower. Currently we are capturing by reference, with `ALPAKA_FN_LAMBDA` a copy will be enforced.
I really would love to separate the capture list from the attributes.

Reply:
host-lambdas may have ref-capture, but not host-device. nvcc will complain about that and `ALPAKA_FN_LAMBDA` is defined as host-device.
Offered solution: 
- host-lambdas are plain-style including capture, only need host-scope binding for HIP/HCC
- `ALPAKA_FN_LAMBDA` is for the non-host and includes `[=]`, cuz everything else would not compile with nvcc
- documentation on this updated